Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/BaseLinearElasticityFEMForceField.h

```diff
@@ -38,27 +38,53 @@ class BaseLinearElasticityFEMForceField : virtual public core::behavior::ForceFi
 
     SOFA_CLASS(SOFA_TEMPLATE(BaseLinearElasticityFEMForceField, DataTypes), SOFA_TEMPLATE(core::behavior::ForceField, DataTypes));
 
-    Data<Real> d_poissonRatio; ///< FEM Poisson Ratio in Hooke's law [0,0.5[
+    Data<VecReal > d_poissonRatio; ///< FEM Poisson Ratio in Hooke's law [0,0.5[
     Data<VecReal > d_youngModulus; ///< FEM Young's Modulus in Hooke's law
 
     /// Link to be set to the topology container in the component graph.
     SingleLink<BaseLinearElasticityFEMForceField<DataTypes>, sofa::core::topology::BaseMeshTopology, BaseLink::FLAG_STOREPATH|BaseLink::FLAG_STRONGLINK> l_topology;
 
-    static inline const VecReal defaultYoungModulusValue = []()
-    {
-        VecReal newY;
-        newY.resize(1);
-        newY[0] = 5000;
-        return newY;
-    }();
-
     BaseLinearElasticityFEMForceField();
+
     void init() override;
 
     void setPoissonRatio(Real val);
     void setYoungModulus(Real val);
 
     Real getYoungModulusInElement(sofa::Size elementId);
+    Real getPoissonRatioInElement(sofa::Size elementId);
+
+    struct ElementsType2D{};
+    struct ElementsType3D{};
+
+    static constexpr ElementsType2D elementsType2D;
+    static constexpr ElementsType3D elementsType3D;
+
+    static std::pair<Real, Real> toLameParameters(ElementsType2D, Real youngModulus, Real poissonRatio);
+    static std::pair<Real, Real> toLameParameters(ElementsType3D, Real youngModulus, Real poissonRatio);
+
+protected:
+
+    static constexpr Real defaultYoungModulusValue = 5000;
+    static inline const VecReal defaultVecYoungModulusValue = []()
+    {
+        VecReal newY;
+        newY.resize(1);
+        newY[0] = defaultYoungModulusValue;
+        return newY;
+    }();
+
+    static constexpr Real defaultPoissonRatioValue = 0.45;
+    static inline const VecReal defaultVecPoissonRatioValue = []()
```

Comment:
Using the call on lambda to initialize the defaultVecPoissonRatioValue looks a bit un-usual. 
Maybe a small text of code comment is needed here (especially why it is different to 
```cpp
    static inline const VecReal defaultVecPoissonRatioValue {defaultYoungModulusValue};
```

Reply:
It's a not-so-new technique to initialize a complex const variable (https://herbsutter.com/2013/04/05/complex-initialization-for-a-const-variable/). However, in this case, it is not so complex as your suggestion simplifies the initial code. I took example from the original code, and did not think about simplifying it.

Reply:
Actually, your suggestion requires https://github.com/sofa-framework/sofa/pull/4868 to compile SofaCUDA.